go-controller/pkg/node/node.go

```diff
@@ -478,6 +501,62 @@ func (n *OvnNode) Start(wg *sync.WaitGroup) error {
 	return err
 }
 
+//AddOrRemoveTaint adds or removes the network-unavailable taint on the node depending on the setTaint value
+func (n *OvnNode) AddOrRemoveTaint(setTaint bool) error {
+	action := ""
+	if setTaint {
+		action = "adding"
+	} else {
+		action = "removing"
+	}
+
+	node, err := n.Kube.GetNode(n.name)
+	networkUnavailableTaint := &kapi.Taint{Key: "k8s.ovn.org/network-unavailable", Value: "NoSchedule", Effect: "NoSchedule"}
+
+	if err != nil {
+		klog.Infof("Unable to retrieve node %s for %s taint: %v", n.name, action, err)
+		return err
+	}
+
+	nodeObjJson, err := json.Marshal(node)
+	if err != nil {
+		klog.Infof("Unable to marshal original node %s object for %s taint: %v", n.name, action, err)
+		return err
+	}
+
+	var taintedNodeObj *kapi.Node
+	if setTaint {
+		taintedNodeObj, _, err = taints.AddOrUpdateTaint(node, networkUnavailableTaint)
+	} else {
+		taintedNodeObj, _, err = taints.RemoveTaint(node, networkUnavailableTaint)
+	}
+
+	if err != nil {
+		klog.Infof("Unable to (un)taint node %s: %v", n.name, err)
+		return err
+	}
+
+	taintedNodeObjJson, err := json.Marshal(taintedNodeObj)
+	if err != nil {
+		klog.Infof("Unable to marshal updated node %s object for %s taint: %v", n.name, action, err)
+		return err
+	}
+
+	patchBytes, err := strategicpatch.CreateTwoWayMergePatch(nodeObjJson, taintedNodeObjJson, kapi.Node{})
+	if err != nil {
+		klog.Infof("Unable to patch the updated node %s object for %s taint: %v", n.name, action, err)
+		return err
+	}
+
+	if _, err = n.client.CoreV1().Nodes().Patch(context.TODO(), n.name, apimachinerytypes.StrategicMergePatchType, patchBytes, metav1.PatchOptions{}); err != nil {
+		klog.Infof("Unable to patch the updated node %s object for %s taint: %v", n.name, action, err)
+		return err
+	}
```

Comment:
in #2326 I am introducing https://github.com/ovn-org/ovn-kubernetes/blob/45f51f0cc8ee83c19f61bec89d568588055f0ee0/go-controller/pkg/kube/kube.go#L223 and https://github.com/ovn-org/ovn-kubernetes/blob/45f51f0cc8ee83c19f61bec89d568588055f0ee0/go-controller/pkg/kube/kube.go#L234 which could be used for this.

Reply:
ty! I am copying the change and will send out the PR shortly!

Reply:
Hey @tssurya,
you don't need to copy the change. You can simply cherry-pick the commit. Otherwise I can simply refactor your code after I got your changes for the taint packages.

Reply:
@creydr : no that's not what I meant :) I have already copied the code from k8s taints package and added it to ovn-k. We can both use those functions and now do not need to vendor k8s.

Reply:
I am testing this on OCP, once it passes will push it here.